jina/executors/indexers/__init__.py

```diff
@@ -196,32 +196,13 @@ def flush(self):
         except:
             pass
 
-    def _filter_nonexistent_keys_values(self, keys: Iterable, values: Iterable, existent_keys: Iterable,
-                                        check_path: str) -> Tuple[List, List]:
-        keys = list(keys)
-        values = list(values)
-        if len(keys) != len(values):
-            raise ValueError(f'Keys of length {len(keys)} did not match values of length {len(values)}')
-        indices_to_drop = self._get_indices_to_drop(keys, existent_keys, check_path)
-        keys = [keys[i] for i in range(len(keys)) if i not in indices_to_drop]
-        values = [values[i] for i in range(len(values)) if i not in indices_to_drop]
-        return keys, values
-
-    def _filter_nonexistent_keys(self, keys: Iterable, existent_keys: Iterable, check_path: str) -> List:
-        keys = list(keys)
-        indices_to_drop = self._get_indices_to_drop(keys, existent_keys, check_path)
-        keys = [keys[i] for i in range(len(keys)) if i not in indices_to_drop]
-        return keys
-
-    def _get_indices_to_drop(self, keys: List, existent_keys: Iterable, check_path: str):
-        indices_to_drop = []
-        for key_index, key in enumerate(keys):
-            if key not in existent_keys:
-                indices_to_drop.append(key_index)
-        if indices_to_drop:
-            self.logger.warning(
-                f'Key(s) {[keys[i] for i in indices_to_drop]} were not found in {check_path}. Continuing anyway...')
-        return indices_to_drop
+    def _filter_nonexistent_keys_values(self, keys: Iterable, values: Iterable, existent_keys: Iterable) -> Tuple[
+        Iterable, Iterable]:
+        filtered_list = [[key, value] for key, value in zip(keys, values) if key in existent_keys]
```

Comment:
Should we not still check the length of the two match?

Reply:
`zip()` swallows that problem

Reply:
In what scenario would we have different lengths?

Reply:
Don't remember the exact reason it came up. But we had that check before to fix some issue that I can't recall

Reply:
Okay, let me check if I find out if the reason still exists.

Reply:
note, all sanity check should be done at the driver level

Reply:
Before my refactoring of the cache, the cache was using `_filter_nonexistent_keys_values` as well. 
Could it be related to the previous data structure in the cache?

Reply:
We can't do the check if keys exist in the driver because we don't have access to the existing keys, do we?